### DISCUSSION TITLE
Ability to Eager load Purchasables on Order Line Items

### DISCUSSION THREAD

**Discussion by hahnzilla** [craftcms/commerce#2396] [Ideas]
**What are you trying to do?**
Something like this has the potential to create a high number of queries. Especially if you're getting multiple orders
```
        foreach ($order->getLineItems() as $item) {
            $item->purchasable->product->hsCode;
        }
```

A few other items are being added to allow for eager loading for records related to an order, but this would be something that's related to a relation of an order.

Luke explains it way better than I could here: https://github.com/craftcms/commerce/issues/1603#issuecomment-667158303 
 
**What's your proposed solution?**
Perhaps `withLineItems()` can take an arg of what it can eager load? Something like `withLineItems(['purchasable', 'purchasable.product', 'purchasable.product.featuredImage'])`

**Additional context**
Originally chatted about here: https://github.com/craftcms/commerce/issues/1603

**Comment by lukeholder** [comment]
To use the standard element relational eager loading of products/purchasables, we would need to make line items elements, which will not be happening in the short-term. The hard part about this is also that purchasable can be of multiple element types, which would be hard to eager load without knowing the type up front (store it on the line item?).

Leaving the ticket open as a feature request.
